lib/src/results.dart

```diff
@@ -207,3 +210,33 @@ extension RealmResultsInternal on RealmResults {
     return RealmResults<T>._(handle, realm);
   }
 }
+
+class _RealmResultsIterator<T extends RealmObject> implements Iterator<T> {
+  final RealmResults<T> _results;
+  final int _length;
+  int _index;
+  T? _current;
+
+  _RealmResultsIterator(RealmResults<T> results)
+      : _results = results,
+        _length = results.length,
+        _index = 0;
+
+  @override
+  T get current => _current as T;
```

Comment:
How do you access the first element? Calling `moveNext` when `_index == 0` will set `_current = _results[1]`. Prior to that `_current == 0`.

Reply:
Scratch that :-) .. 

> bool moveNext()
> Advances the iterator to the next element of the iteration.
> 
> Should be called before reading current. If the call to moveNext returns true, then current will contain the next element of the iteration until moveNext is called again. If the call returns false, there are no further elements and current should not be used any more.
> 
> It is safe to call moveNext after it has already returned false, but it must keep returning false and not have any other effect.
> 
> A call to moveNext may throw for various reasons, including a concurrent change to an underlying collection. If that happens, the iterator may be in an inconsistent state, and any further behavior of the iterator is unspecified, including the effect of reading current.

Reply:
On the first call of moveNext index == 0 and  _current  _results[index]. After that index is incremented. It seems to be correct according to me.